src/validation/__tests__/OverlappingFieldsCanBeMergedRule-test.ts

```diff
@@ -613,6 +653,139 @@ describe('Validate: Overlapping fields can be merged', () => {
       ]);
     });
 
+    describe('nullable field on types which potentially overlap', () => {
+      it('matching nullability status', () => {
+        // This is invalid since an object could potentially be both the Object
+        // type IntBox and the interface type NonNullStringBox1. While that
+        // condition does not exist in the current schema, the schema could
+        // expand in the future to allow this. Thus it is invalid.
```

Comment:
I think this comment needs to be updated to describe the test. It looks like this test is expected to succeed, but the comment indicates that it should fail